src/PubkeyEncryptManager.php

```diff
@@ -243,4 +244,42 @@ public function deleteRoleKey(Role $role) {
       ->delete();
   }
 
+  /*
+   * Initialize Encryption Profiles upon module installation.
+   */
+  public function initializeEncryptionProfiles() {
+    // Load all Role keys.
+    $keys = \Drupal::service('key.repository')
+      ->getKeysByProvider('pubkey_encrypt');
+
+    foreach ($keys as $key){
+      $this->generateEncryptionProfile($key);
+    }
+  }
+
+  /*
+   * Generate an Encryption profile for a Role key.
+   */
+  public function generateEncryptionProfile(Key $key) {
```

Comment:
@param is missing from the docblock above.  If this is the same as the parent, then @inheritdoc can be used.